From 7eff8f89b32b6921a0d86c50c6c62154c6ddc96e Mon Sep 17 00:00:00 2001
From: Steve Langasek <steve.langasek@ubuntu.com>
Date: Fri, 2 Jun 2023 16:30:19 +0000
Subject: [PATCH 3/3] Also emit -Werror=implicit-function-declaration for
 feature=+time64

Per https://lists.debian.org/debian-devel/2023/05/msg00262.html et al.,
missing glibc includes can cause packages to link to the wrong symbols,
potentially causing crashes or misbehavior.  Since functions that use
time_t are fairly ubiquitous, there's a high risk of this happening for
*some* package in Debian.  Better to make all software with missing
function declarations fail to build now, than to spend all cycle tracking
down runtime bugs.
---
 scripts/Dpkg/Vendor/Debian.pm | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/scripts/Dpkg/Vendor/Debian.pm b/scripts/Dpkg/Vendor/Debian.pm
index 20d77fea1..803949024 100644
--- a/scripts/Dpkg/Vendor/Debian.pm
+++ b/scripts/Dpkg/Vendor/Debian.pm
@@ -400,6 +400,8 @@ sub _add_build_flags {
 
     if ($flags->use_feature('feature', 'time64')) {
         $flags->append('CPPFLAGS', '-D_TIME_BITS=64');
+        $flags->append('CFLAGS', '-Werror=implicit-function-declaration');
+        $flags->append('CXXFLAGS', '-Werror=implicit-function-declaration');
     }
 
     ## Area: qa
-- 
2.40.1

